kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/records/KafConsumerRecord.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.clients.records;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.systemtests.utils.KafkaUtils;
+
+/**
+ * The type Kaf consumer record.
+ */
+public class KafConsumerRecord extends BaseConsumerRecord {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafConsumerRecord.class);
+
+    private String timestamp;
+    private String timestampType;
+    private List<Map<String, String>> headers;
+
+    /**
+     * Sets timestamp.
+     *
+     * @param timestamp the timestamp
+     */
+    public void setTimestamp(String timestamp) {
+        this.timestamp = timestamp;
+    }
+
+    /**
+     * Sets timestamp type.
+     *
+     * @param timestampType the timestamp type
+     */
+    public void setTimestampType(String timestampType) {
+        this.timestampType = timestampType;
+    }
+
+    /**
+     * Sets headers.
+     *
+     * @param headers the headers
+     */
+    public void setHeaders(List<Map<String, String>> headers) {
+        this.headers = headers;
+    }
+
+    /**
+     * Parse from json string.
+     *
+     * @param response the response
+     * @return the kaf consumer record
+     */
+    public static KafConsumerRecord parseFromJsonString(String response) {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        try {
+            return objectMapper.readValue(response, KafConsumerRecord.class);
+        }
+        catch (JsonProcessingException e) {
+            LOGGER.atError().setMessage("Error: {}").addArgument(e.getMessage()).log();
```

Comment:
you should almost always chain the exception.

Reply:
Do you mean throw the exception? I don't want to do that as we can found some logs that are not just the JSON that we are interested in, and I don't want to fail because of that, we must go on with the next line received

Reply:
I used the wrong word.  I meant pass the exception object to the logger so it will log the full stack trace.  If you just log the exception's message, a developer has less content to understand the issue.